src/aspire/operators/filters.py

```diff
@@ -241,25 +244,43 @@ def __init__(self, xfer_fn_array):
         self.xfer_fn_array = xfer_fn_array
 
     def _evaluate(self, omega):
-        sz = self.sz
-        # TODO: This part could do with some documentation - not intuitive!
-        temp = np.array(sz)[:, np.newaxis]
-        omega = (omega / (2 * np.pi)) * temp
-        omega += np.floor(temp / 2) + 1
-
-        # Emulating the behavior of interpn(V,X1q,X2q,X3q,...) in MATLAB
-        _input_pts = tuple(list(range(1, x + 1)) for x in self.xfer_fn_array.shape)
-        interpolator = RegularGridInterpolator(
-            _input_pts, self.xfer_fn_array, bounds_error=False, fill_value=0
-        )
-        result = interpolator(
-            # Split omega into input arrays and stack depth-wise because that's how
-            # the interpolator wants it
-            np.dstack(np.split(omega, len(sz)))
-        )
 
-        # Result is 1 x np.prod(sz) in shape; convert to a 1-d vector
-        result = np.squeeze(result, 0)
+        _input_pts = tuple(np.linspace(1, x, x) for x in self.xfer_fn_array.shape)
+
+        if self.sz == self.xfer_fn_array.shape:
```

Comment:
This doesn't guarantee that the grid values are the same, unfortunately. I think the best way to do this short-circuiting is to override the `evaluate_grid` method.

Reply:
Okay yeah, I think I was able to implement this optimization via `evaluate_grid`.  The path was a little tricky because the evaluate_grid call was called from the PowerFilter, not the ArrayFilter...  This should only trip via `evaluate_grid` now, so other invocations of ArrayFilter evaluation will still go through the interp as they were.

Checkout the recent commit and let me know if that looks better.